### ISSUE TITLE
Unwanted assigning to free variable

### ISSUE THREAD

**Issue by DamienCassou** [emacs-elsa/Elsa#113]
```emacs
;;; foo.el --- Coucou test                           -*- lexical-binding: t; -*-

(defvar foo-bar nil "Som documentation.")

(defun foo-do ()
  (setq foo-bar t))

(provide 'foo)
;;; foo.el ends here
```

Elsa will complain on the `setq` line that we are assigning to a free variable. But the variable is defined with `defvar`.

**Comment by Fuco1** [comment]
I was pretty sure this actually worked already.  Definitely a bug.

**Comment by Fuco1** [comment]
Fixed in 7399271
